doc/architecture/domains/authorization/features/authorize-user.md

```diff
@@ -0,0 +1,210 @@
+﻿# User Consent Authorization Documentation
+
+## Overview
+
+The User Consent Authorization API is responsible for retrieving and managing user authorization models.
+This endpoint is designed to work in tandem with the Azure B2C tenant,
+which is the only authorized caller of this endpoint.
+The primary purpose is to validate user consent, check terms acceptance,
+and manage user and organization relationships in the system.
+
+## User Consent Flow
+
+```mermaid
+sequenceDiagram
+    participant B2C as Azure B2C
+    participant API as Authorization API
+    participant Handler as Command Handler
+    participant DB as Database
+
+    activate B2C
+    B2C->>API: POST /api/authorization/user-consent/
+    Note over B2C,API: Send user info (sub, name, orgName, orgCvr)
+    activate API
+
+    API->>Handler: Create and send GetConsentForUserCommand
+    Note over API,Handler: Command contains sub, name, orgName, orgCvr
+    activate Handler
+
+    rect rgb(50, 50, 50)
+        Note over Handler, DB: Database Interactions
+        Handler->>DB: Begin Transaction
+        activate DB
+
+        Handler->>DB: Query Organization by CVR
+        DB-->>Handler: Return Organization (if exists)
+
+        alt Organization doesn't exist
+            Handler->>DB: Create new Organization
+            DB-->>Handler: Return new Organization
+        end
+
+        Handler->>DB: Query Latest Terms
+        DB-->>Handler: Return Latest Terms
+
+        Handler->>DB: Query User by Sub
+        DB-->>Handler: Return User (if exists)
+
+        alt User doesn't exist
+            Handler->>DB: Create new User
+            DB-->>Handler: Return new User
+        end
+
+        Handler->>DB: Check/Update User-Organization Affiliation
+        DB-->>Handler: Confirm Affiliation
+
+        alt Terms not accepted or outdated
+            Handler->>DB: Update Terms Acceptance Status
+            DB-->>Handler: Confirm Update
+        end
+
+        Handler->>DB: Commit Transaction
+        DB-->>Handler: Confirm Commit
+        deactivate DB
+    end
```

Comment:
Why the black box? Can't see what's inside

Reply:
Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1279/commits/5f203707232534e753d4742b5869859da732d5cc